Remove topics and groups "blacklist" pattern from KafkaMirrorMaker2 CRD (#4988)
strimzi/strimzi-kafka-operator: documentation/modules/appendix_crds.adoc

|xref:type-KafkaMirrorMaker2ConnectorSpec-{context}[`KafkaMirrorMaker2ConnectorSpec`]
|topicsPattern           1.2+<.<a|A regular expression matching the topics to be mirrored, for example, "topic1\|topic2\|topic3". Comma-separated lists are also supported.
|string

> |topicsBlacklistPattern  1.2+<.<a|*The `topicsBlacklistPattern` property has been deprecated, and should now be configured using `topicsExcludePattern`.* A regular expression matching the topics to exclude from mirroring. Comma-separated lists are also supported.

Reviewer: You will need to rebuild the API documentation.

> |topicsBlacklistPattern  1.2+<.<a|*The `topicsBlacklistPattern` property has been deprecated, and should now be configured using `topicsExcludePattern`.* A regular expression matching the topics to exclude from mirroring. Comma-separated lists are also supported.

Author: @laidan6000 I think these descriptions are auto generated from the api module.

> |topicsBlacklistPattern  1.2+<.<a|*The `topicsBlacklistPattern` property has been deprecated, and should now be configured using `topicsExcludePattern`.* A regular expression matching the topics to exclude from mirroring. Comma-separated lists are also supported.

Author: @laidan6000 I think these descriptions are auto generated from the api module.

> |topicsBlacklistPattern  1.2+<.<a|*The `topicsBlacklistPattern` property has been deprecated, and should now be configured using `topicsExcludePattern`.* A regular expression matching the topics to exclude from mirroring. Comma-separated lists are also supported.

Reviewer 2: TBH, I think we should keep it as it is. This is the generated default for all the fields we have.

> |topicsBlacklistPattern  1.2+<.<a|*The `topicsBlacklistPattern` property has been deprecated, and should now be configured using `topicsExcludePattern`.* A regular expression matching the topics to exclude from mirroring. Comma-separated lists are also supported.

Reviewer: But I thought all descriptions could be changed by editing the Java files in the `api/` folder?

> |topicsBlacklistPattern  1.2+<.<a|*The `topicsBlacklistPattern` property has been deprecated, and should now be configured using `topicsExcludePattern`.* A regular expression matching the topics to exclude from mirroring. Comma-separated lists are also supported.

Author: @laidan6000 I could remove the property description from deprecated properties, but changing the deprecation message would be out of scope for this PR as it would affect all deprecated properties.

> |topicsBlacklistPattern  1.2+<.<a|*The `topicsBlacklistPattern` property has been deprecated, and should now be configured using `topicsExcludePattern`.* A regular expression matching the topics to exclude from mirroring. Comma-separated lists are also supported.

Reviewer 2: But why should the deprecated properties have no description? Users will be still using them so I think keeping the description makes sense ... that is also what we did for all other deprecated properties in the past.

> |topicsBlacklistPattern  1.2+<.<a|*The `topicsBlacklistPattern` property has been deprecated, and should now be configured using `topicsExcludePattern`.* A regular expression matching the topics to exclude from mirroring. Comma-separated lists are also supported.

Author: Yep, I agree @scholzj.

> |topicsBlacklistPattern  1.2+<.<a|*The `topicsBlacklistPattern` property has been deprecated, and should now be configured using `topicsExcludePattern`.* A regular expression matching the topics to exclude from mirroring. Comma-separated lists are also supported.

Reviewer: Thanks for the explanation. I didn't realise that this would affect all the deprecated properties.

|string
|topicsExcludePattern    1.2+<.<a|A regular expression matching the topics to exclude from mirroring. Comma-separated lists are also supported.
|string
|groupsPattern           1.2+<.<a|A regular expression matching the consumer groups to be mirrored. Comma-separated lists are also supported.
|string

> |groupsBlacklistPattern  1.2+<.<a|*The `groupsBlacklistPattern` property has been deprecated, and should now be configured using `groupsExcludePattern`.* A regular expression matching the consumer groups to exclude from mirroring. Comma-separated lists are also supported.

Reviewer: ```suggestion
|groupsBlacklistPattern  1.2+<.<a|Deprecated. Configure using the `groupsExcludePattern` property instead.
```

Reviewer: You will need to rebuild the API documentation.

Author: @laidan6000 I think these descriptions are auto generated from the api module.

Author: @laidan6000 I think these descriptions are auto generated from the api module.
